persist/mongodb/user.go

```
@@ -117,11 +114,8 @@ func (u *UserMongoRepository) GetByAddress(pCtx context.Context, pAddress string
 		return nil, err
 	}
 
-	if len(result) == 0 {
-		return nil, fmt.Errorf("no users found")
-	}
-	if len(result) > 1 {
-		return nil, fmt.Errorf("more than one user found when expecting a single result")
+	if len(result) != 1 {
+		return nil, persist.ErrUserNotFoundByAddress{Address: pAddress}
```

Any chance `result` can be > 1 ? It technically shouldn't happen since addresses should be unique to a user, but is there any coded constraint that guarantees `result` will never be > 1? Maybe there's something about the mongo find that I'm missing.

I do see you originally had a condition for if there was more than 1 user found, do we not need it anymore?

---

For most cases (like get by ID requests) we can always guarantee 1 result and for something like addresses with the way our current codebase looks it would also be impossible to get more than one (of course this might change in the future if we want the same wallet to be able to be in multiple accounts).  I think checking if the result is more than one might be useful information for us (that means we did something wrong somewhere else in our code) but I don't think it would be useful information for the user. 

We could add an extra check that sees if it returns more than one and then logs it so that we can fix it or if you think its best we can return a different error to the user.